improve navigation

rccsousa opened the pull request:
why:
- author pills were not linking to author pages, 
- content cards were linking to content on their entirety thus rendering buttons unclicable, 
- highlights were not linking to content;
how:
- refactored Author pill to conditionally render based on number of authors;
- added links to elements;
- adress category styling on hub highlights, solving #45

rccsousa committed:
update: convert to clickable link to author page

rccsousa committed:
fix: change featured image container position to relative

rccsousa committed:
refactor Authors component and AuthorPills

- created pills for 2 authors and 3+ authors;
- moved Author pills to a single file located in Authors/AuthorPills;
- fixed imports

rccsousa committed:
convert single author pill to a clickable link

rccsousa committed:
make highlighted content clicable and link to content

rccsousa committed:
fix: image container was rendering even without images

rccsousa committed:
fix: image container was rendering even without images

rccsousa committed:
fix styling

rccsousa committed:
fix broken links

rccsousa committed:
make content card link hug featured image

rccsousa committed:
file renaming

rccsousa committed:
update header styling

- make header correspond to page type, changing background and font
  colors based on the type of page it is inserted instead of being
  generic

rccsousa committed:
optimize css

rccsousa committed:
update: improve header dynamic style

rccsousa committed:
update: add socials collection and include social links in footer

rccsousa committed:
fix typescript errors

rccsousa committed:
update metadata generation

- changed metadata generation, badges, favicons and open graphs,
  replacing payload boilerplate open graph and use content tailored to
  the content-sub project

rccsousa commented:
@davelange i'll mark this as ready for review for now and move on to other smaller adjustments on a new PR as this one is already quite long, but navigation had many things needing fixing;

will do a clean up pass soon

davelange commented on src/app/_components/Authors/AuthorPills/index.tsx:
```diff
@@ -0,0 +1,72 @@
+  );
+}
+
+export function TwoAuthorPill({ authors }) {
```
It would be easier if there's only 1 AuthorPill. You could continue having the TwoAuthorPill and VariousAuthorsPill, but I'd suggest only exporting one AuthorPill that receives a user array, checks the length, and internally decides if it should use SingleAuthor, TwoAuthor, or Various Author. That way when you use it elsewhere, you dont need to worry about how which version to use

rccsousa replied:
Great idea, i'll implement this, thanks

rccsousa replied:
done in https://github.com/subvisual/content-sub/pull/58/commits/90ef400d24d1cbf97c7e636fcdc9d003a8907884 ✅

rccsousa committed:
refactor AuthorPill

- remake author pill as a standalone component;
- use conditional rendering to make it fit one, two or more
contributors;

rccsousa committed:
fix: refactor to reflect new authorpill changes
